src/praisonai-agents/praisonaiagents/compaction/compactor.py

```
@@ -123,11 +190,80 @@ def compact(
+        processed_messages = []
+        pruned_count = 0
+        
+        for msg in messages:
+            role = msg.get("role", "")
+            content = msg.get("content", "")
+            
+            # Process tool result messages
+            if role == "tool" or msg.get("tool_call_id"):
+                if isinstance(content, str) and content:
+                    # Create hash of tool result content
+                    content_hash = hashlib.md5(content.encode()).hexdigest()[:8]
```

Ruff (0.15.15)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)